### PR TITLE
React/PT-69/Tweets list

### PR DESCRIPTION
**Pull request by danybony**
This PR adds two components shared for both iOS and Android:

- a tweets list component, which in the constructor initialises a mock set of data with a structure similar to the real Twitter data for the home tweet feed
- a single tweet item component, which displays some of the tweet data given a single row passed from the list

Screenshots: 

| iOS | Android |
| --- | --- |
| screen shot 2016-05-11 at 12 35 26 | screen shot 2016-05-11 at 12 35 06 |

Paired with: 
@ me

### CODE REVIEW

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react'
+import { ListView } from 'react-native'
+import TweetsListItem from './tweetsListItem'
+
+class TweetsList extends Component {
+
+  constructor (props) {
+    super(props)
+    console.log('constructor')
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: lol

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: `!==` 👍 like a pro

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+  componentDidMount () {
+    console.log('componentDIdMount')
+    this.refreshData()
+    console.log(this.state)
+  }
+
+  refreshData () {
+    this.setState({
+      dataSource: this.state.dataSource.cloneWithRows([
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: To save horizontal space, I usually align the first `{` with the array bracket, so you can align the following lines with 2 less spaces.

**Comment by frapontillo** [comment]
Other than stupid aesthetics, looks good 👍

File: ReactNative/ReactTwitter/core/views/tweetsList.js (hunk 3 above)
Comment (danybony) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: 👎  It doesn't look more readable as you suggested, it actually looks worse to me.

Comment (frapontillo) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: Boooooo

**Comment by eduardb** [comment]
:shipit:

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    var url = 'https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=' + userHandle
+    fetch(url, {
+      method: 'GET',
+      headers: {
+        'Authorization': 'OAuth oauth_consumer_key="aqPSTs1FT2ndP7qXi247BtbXd", oauth_nonce="987e33536527c50a7a0e1eb7b2d77e36", oauth_signature="K5W7yxsVPmQgd8arTIJ8qZXq%2FTk%3D", oauth_signature_method="HMAC-SHA1", oauth_timestamp="1463065323", oauth_token="730013266697654273-RDssoTCOdHNQtFA9k87OSijeZHcF4SU", oauth_version="1.0"'
+      }
+    })
+      .then((response) => response.json())
+      .then((rjson) => {
+        console.log(rjson)
```
Comment (gbasile) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: do we need this log?

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  componentDidMount () {
+    this._refreshData()
+  }
+
+  /*global fetch*/
+  /*eslint no-undef: "error"*/
+  _refreshData () {
+    var userHandle = 'twitterapi'
+    var url = 'https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=' + userHandle
+    fetch(url, {
```
Comment (frapontillo) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: Also, remember to use `NetworkCallsManager` when #35 is merged.

File: ReactNative/ReactTwitter/core/views/tweetsList.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+import { ListView } from 'react-native'
+import TweetsListItem from './tweetsListItem'
+
+class TweetsList extends Component {
+
+  constructor (props) {
+    super(props)
+    var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2})
+    this.state = {
+      dataSource: ds.cloneWithRows([])
```
Comment (gbasile) [inline ReactNative/ReactTwitter/core/views/tweetsList.js]: Can you explain a little bit how this cloneWithRows actually works?

File: ReactNative/ReactTwitter/core/views/tweetsListItem.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  },
+
+  _formatTime (time) {
+    var dateFormat = require('dateformat')
+    return dateFormat(time, 'mmmm dS, yyyy, h:MM:ss TT')
+  },
+
+  _tweetSelected (tweetId) {
+    // TODO: navigate to tweet details screen
+    Alert.alert(
```
Comment (gbasile) [inline ReactNative/ReactTwitter/core/views/tweetsListItem.js]: It's interesting to see which iOS component is being used for the alert as we have a couple of them and one is going to be deprecated

**Pull request merged**